### PR TITLE
in - Bug Fix Remove +/- Button for Basic Course Search where not needed

### LINKED ISSUE #7: BUG - Courses without sections have a + button

### PR DESCRIPTION
**Pull request by nagpalishaan**
Updated SectionsTableBase.js to remove the button that shows +/- when there is no associated sections with it.

Note: Used rows.subRows.length > 1 instead of canExpand/depth and other implementations since they don't work as intended.
Note: rows.subRows.length > 1 because each row has 1 empty subRow 

# Storybook
- [Before](https://ucsb-cs156-s23.github.io/proj-courses-s23-7pm-3/storybook/?path=/docs/components-sections-sectionstable--empty)
- [After](https://ucsb-cs156-s23.github.io/proj-courses-s23-7pm-3/prs/37/storybook/?path=/docs/components-sections-sectionstable--empty)

# Screenshots
### Before
MATH 101B has no sections but the plus button can be clicked:
image

### After
Classes without sections no longer have the button:
image

### CODE REVIEW

**Comment by nagpalishaan** [comment]
> Updated SectionsTableBase.js to remove the button that shows +/- when there is no associated sections with it.
> 
> Note used rows.subRows.length > 1 instead of canExpand/depth and other implementations since they don't work as intended. Note rows.subRows.length > 1 because each row has 1 empty subRow

It is now rows.subRows.length >= 2 to prevent mutations

**Review by KaneD985 — COMMENTED** [review]
LGTM

**Review by KaneD985 — DISMISSED** [review]
LGTM (I'll approve this after the screenshots are added)

**Review by wq12li — CHANGES_REQUESTED** [review]
The functionality seems to work both locally and on qa, which is great.
I'd just like to ask you to add some screenshots to your PR, so that other reviewers can see what it's supposed to look like at a glance.

**Review by wq12li — DISMISSED** [review]
I went ahead and edited the PR to have Storybook links and screenshots.

File: frontend/src/main/components/SectionsTableBase.js
Comment (pconrad) [inline frontend/src/main/components/SectionsTableBase.js]: I like the use of the ? operator.  I wonder, though, if this would be more clear to future programmers if we included some (redundant) parens to show the operator order?  Something like this:
```suggestion
                    {  ( row.subRows.length > 1  ) ? ( row.isExpanded ? "➖ " : "➕ " ) : null} 
```

File: frontend/src/main/components/SectionsTableBase.js
Comment (pconrad) [inline frontend/src/main/components/SectionsTableBase.js]: It's debatable whether this comment should stay in the code or not.    I'll let you decide, but I wanted to at least invite you to ponder it.

**Review by pconrad — CHANGES_REQUESTED** [review]
A couple of small suggestions; otherwise, looks great!

File: frontend/src/main/components/SectionsTableBase.js
Comment (nagpalishaan) [reply frontend/src/main/components/SectionsTableBase.js]: I have removed the comment as it isn't too necessary.

File: frontend/src/main/components/SectionsTableBase.js
Comment (pconrad) [inline frontend/src/main/components/SectionsTableBase.js]: Why are we disabling all here?

All of this looks very testable.

Stryker disable is only for cases of very complex logic (such as the interaction of the various caches inside of the "react-query" library, and local storage. (Even local storage and react-query we now have solutions for, so technically, there should be no *new* code that falls under stryker exceptions.).

Please remove this and let's see if we can deal with whatever Stryker is throwing at us.  This is just sweeping it under the rug.

**Review by pconrad — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
